### PR TITLE
Migrate to Redux. Functional components, prop-types.

### PR DESCRIPTION
**Pull request by Andrii-D**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -7,12 +7,14 @@
     "@blueprintjs/core": "^3.6.1",
     "@material-ui/core": "^3.1.1",
     "gh-pages": "^2.0.0",
+    "prop-types": "^15.6.2",
     "react": "^16.5.2",
```
Comment (oleg-preply) [inline package.json:11]: keep prop-types same version as react :wink:

File: package.json [hunk 2]
```diff
@@ -28,7 +30,8 @@
     "eslint-config-airbnb": "^17.1.0",
     "eslint-plugin-import": "^2.14.0",
     "eslint-plugin-jsx-a11y": "^6.1.1",
-    "eslint-plugin-react": "^7.11.1"
+    "eslint-plugin-react": "^7.11.1",
+    "prettier": "1.14.3"
```
Comment (oleg-preply) [inline package.json:34]: :fire:

File: src/App.js [hunk 3]
```diff
@@ -1,116 +1,151 @@
...
   {device: 'iPhone 7+', os: 'iOS', width: 1080, height: 1920},
   {device: 'Samsung Galaxy S6', os: 'Android', width: 1440, height: 2560},
   {device: 'Google Pixel', os: 'Android', width: 1800, height: 1920},
   {device: 'iPhone 4', os: 'iOS', width: 640, height: 960},
   {device: 'Nokia Lumia', os: 'Windows', width: 480, height: 800},
   {device: 'iPhone 3G', os: 'iOS', width: 320, height: 480},
-].map(item => { return {ratio: (item.height/item.width).toFixed(2), ...item}})
+].map (item => {
+  return {ratio: (item.height / item.width).toFixed (2), ...item};
+});
```
Comment (oleg-preply) [inline src/App.js]: oh I love this way (array & map) of declaring constants :heart_eyes: 

p.s.  could be `item => ({ratio: (item.height / item.width).toFixed (2), ...item}}` - eslint should tell you that (it's all over the codebase :))

Comment (oleg-preply) [reply src/App.js]: `toFixed (2)` - no space

File: src/App.js [hunk 4]
```diff
@@ -1,116 +1,151 @@
...
 
-const LANGUAGES = ['en', 'ua', 'ru', 'es', 'pl', 'fr']
+export const LANGUAGES = ['en', 'ua', 'ru', 'es', 'pl', 'fr'];
 
-let ScreenItem = ({ item }) => (
+let ScreenItem = ({item}) => (
   <span>
     <strong>{item.device}</strong>
-    {" " + item.width + "x" + item.height}
+    {' ' + item.width + 'x' + item.height}
```
Comment (oleg-preply) [inline src/App.js]: can use template literal - \`string${var}string\`

File: src/Screen.js [hunk 5]
```diff
@@ -49,27 +50,32 @@ const StyledIframe = styled.iframe`
...
-        );
-    }
-}
-Screen.defaultProps = {
-    src: "http://example.com",
-    item: {device: 'iPhone 7', os: 'iOS', width: 750, height: 1334}
-}
+const Screen = ({scale = 1, src, item}) => {
+  let scaledWidth = item.width * scale / 100 + 32 + 'px';
+  let scaledHeight = item.height * scale / 100 + 120 + 'px';
```
Comment (oleg-preply) [inline src/Screen.js]: it usually better to prefer `const` over the `let`

File: src/Screen.js [hunk 6]
```diff
@@ -1,5 +1,6 @@
-import React, { Component } from 'react';
-import styled from 'styled-components'
+import React from 'react';
+import styled from 'styled-components';
+import PropTypes from 'prop-types';
```
Comment (oleg-preply) [inline src/Screen.js]: having it as `import { func, string, ...etc } from 'prop-types';` can save you some time

File: src/actions.js [hunk 7]
```diff
@@ -0,0 +1,10 @@
+import {SET_LANGUAGES, SET_SCALE, SET_SCREENS, SET_URL, RESET_STATE} from './action-types';
+
+export const setLanguages = languages => ({
+  type: SET_LANGUAGES,
+  payload: languages,
+});
+export const setScreens = screens => ({type: SET_SCREENS, payload: screens});
+export const setUrl = url => ({type: SET_URL, payload: url});
+export const setScale = scale => ({type: SET_SCALE, payload: scale});
+export const resetState = () => ({type: RESET_STATE});
```
Comment (oleg-preply) [inline src/actions.js]: cool! next step is `redux-actions` usage!

File: src/index.css [hunk 8]
```diff
@@ -1,5 +1,12 @@
...
+
 body {
   margin: 0;
   padding: 0;
+  padding-top: 50px;
   font-family: sans-serif;
 }
+.App {
+  text-align: center;
+}
```
Comment (oleg-preply) [inline src/index.css]: with `styled-components` would be more consistent to do it with `injectGlobal` (in v4 they another approach)

File: src/index.js [hunk 9]
```diff
@@ -3,6 +3,18 @@ import ReactDOM from 'react-dom';
 import './index.css';
 import App from './App';
 import registerServiceWorker from './registerServiceWorker';
+import {Provider} from 'react-redux';
```
Comment (oleg-preply) [inline src/index.js:6]: it seems you didn't apply prettier :joy:

File: src/reducers.js [hunk 10]
```diff
@@ -0,0 +1,28 @@
...
+import {LANGUAGES, SCREENS} from './App';
+const initialState = {
+  scale: 50,
+  screens: [SCREENS[0], SCREENS[1]],
+  languages: [LANGUAGES[0], LANGUAGES[1]],
+  // src: "https://www.pexels.com"
+  src: 'http://www.bbc.com',
+};
+
+const rootReducer = (state = initialState, action) => {
```
Comment (oleg-preply) [inline src/reducers.js]: before trying `redux-actions` try to split `rootReducers` to smaller ones and then combine them with `combineReducers`

**Review by oleg-preply — APPROVED** [review]
:awesome:

File: package.json (hunk 1 above)
Comment (Andrii-D) [reply package.json:11]: @oleg-preply How? There is no such version in prop-types: https://www.npmjs.com/package/prop-types

File: src/App.js (hunk 3 above)
Comment (Andrii-D) [reply src/App.js]: `prettier` does some weird stuff, like spaces

File: src/index.js (hunk 9 above)
Comment (Andrii-D) [reply src/index.js:6]: help me with that

**Pull request merged**